### PR TITLE
Vault

### PR DESCRIPTION
**Pull request by twling**
Vault deployment as part of the minimal pipeline.

This is not the 'dev' mode deployment (turns out that was stateless, lose all secrets on pod destruction).

This is the full production ready deployment of a single vault server, tls certificates enabled on vault and injector services. Deploys ready to function for the cluster.

Only major task left undone is that the 'route' is hardcoded. Need to figure out a way to use some kind of variable in the yaml so that it can be deployed to any generic OCP cluster.

@dwinchell

### CODE REVIEW

File: argo-cd-apps/app-of-apps/minimal.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ spec:
 
   source:
     path: argo-cd-apps/overlays/minimal
-    repoURL: https://github.com/ploigos/openshift-pipelines-quickstart.git
-    targetRevision: main
+    repoURL: https://github.com/twling/openshift-pipelines-quickstart.git
```
Comment (dwinchell) [inline argo-cd-apps/app-of-apps/minimal.yml]: Looks accidental

File: argo-cd-apps/base/pipelines-as-a-service/easymode.yml [hunk 2]
```diff
@@ -11,8 +11,8 @@ spec:
 
   source:
     path: components/pipelines-as-a-service/easymode
-    repoURL: https://github.com/ploigos/openshift-pipelines-quickstart.git
-    targetRevision: main
+    repoURL: https://github.com/twling/openshift-pipelines-quickstart.git
+    targetRevision: vault
```
Comment (dwinchell) [inline argo-cd-apps/base/pipelines-as-a-service/easymode.yml]: same

File: argo-cd-apps/base/pipelines-as-a-service/minimal.yml [hunk 3]
```diff
@@ -11,8 +11,8 @@ spec:
 
   source:
     path: components/pipelines-as-a-service/minimal
-    repoURL: https://github.com/ploigos/openshift-pipelines-quickstart.git
-    targetRevision: main
+    repoURL: https://github.com/twling/openshift-pipelines-quickstart.git
+    targetRevision: vault
```
Comment (dwinchell) [inline argo-cd-apps/base/pipelines-as-a-service/minimal.yml]: same

File: argo-cd-apps/base/vault/vault-init.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  namespace: openshift-gitops
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+
+spec:
+  project: default
+
+  source:
+    path: components/vault-init
+    repoURL: https://github.com/twling/openshift-pipelines-quickstart.git
```
Comment (dwinchell) [inline argo-cd-apps/base/vault/vault-init.yml]: twling here too

File: argo-cd-apps/base/vault/vault.yml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+              #utilize OCP service-ca for certs
+              service.beta.openshift.io/serving-cert-secret-name: vault-agent-injector-tls
+        #server config
+        server:
+          image:
+            repository: "registry.connect.redhat.com/hashicorp/vault"
+            tag: "1.10.0-ubi"
+          route:
+            enabled: true
+            host: vault.apps.cluster-k2wgc.k2wgc.sandbox861.opentlc.com
```
Comment (dwinchell) [inline argo-cd-apps/base/vault/vault.yml:47]: rhpds url here

Comment (dwinchell) [reply argo-cd-apps/base/vault/vault.yml:47]: is there a way to make it refer to something local? if not, we should document updating it

**Review by dwinchell — CHANGES_REQUESTED** [review]

File: argo-cd-apps/base/vault/vault.yml (hunk 5 above)
Comment (twling) [reply argo-cd-apps/base/vault/vault.yml:47]: Needs additional research to figure out a good solution here. Updated README with info on what to update.

**Review by dwinchell — APPROVED** [review]

**Pull request merged**